scripts/Set-Version.ps1

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# .SYNOPSIS
+# Update the version number in the repository.
+# .PARAMETER VersionNumber
+# Version number to set.
+
+param ([parameter(Mandatory=$true)][string] $VersionNumber)
+
+$files = @("resource\ebpf_resource.h", "tools\nuget\ebpf-for-windows.nuspec", "scripts\deploy-ebpf.ps1")
+
+$files | % {
+    $file = $_
+    $content = Get-Content $file -Raw
+    $content = $content.Replace("{version}", $VersionNumber)
```

Comment:
if this replaces inline, it will only work once on a given enlistment.  For CI/CD that's ok, but for local build/deploy that is problematic.  Instead it seems like you want to set content in a copy used for build, but leave the original with "{version}" in it.   And make a local build invoke the Set-Version.ps1 script if the copy doesn't exist.  Just like it's common in cmake systems to have a _foo_.in file that is used to generate a _foo_ file used in the build